File: src/main/java/com/ada/conversation/application/LlmRequestFactory.java
package com.ada.conversation.application;

import com.ada.conversation.application.dto.*;
import com.ada.conversation.manager.ContextManager;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LlmRequestFactory {
  private final ContextManager assembler;

  public LlmRequest create(ChatRequest r, String model) {
    var c = assembler.build(r);
    return new LlmRequest(
        model,
        c.messages(),
        c.tools(),
        0.2,
        1024,
        new LlmRequestMetadata(
            UUID.randomUUID().toString(),
            null,
            java.util.List.of(),
            java.util.List.of(),
            c.selection()));
  }
}

File: src/main/kotlin/com/ada/conversation/application/LlmRequestFactory.kt
package com.ada.conversation.application

import com.ada.conversation.application.dto.ChatRequest
import com.ada.conversation.context.ContextAssembler
import com.ada.conversation.application.dto.LlmRequest
import com.ada.conversation.application.dto.LlmRequestMetadata
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class LlmRequestFactory(
    private val contextAssembler: ContextAssembler,
) {
    fun create(request: ChatRequest, model: String): LlmRequest {
        val context = contextAssembler.build(request)
        return LlmRequest(
            model = model,
            messages = context.messages,
            tools = context.tools,
            metadata = LlmRequestMetadata(UUID.randomUUID().toString()),
        )
    }
}
